PSReadLine/Completion.cs

```diff
@@ -429,29 +458,87 @@ private void PossibleCompletionsImpl(CommandCompletion completions, bool menuSel
                 int selectedItem = 0;
                 bool undo = false;
 
-                DoReplacementForCompletion(matches[0], completions);
+                int savedUserMark = _singleton._mark;
```

Comment:
This is a non-static method, so you should use `_mark` instead of `_singleton._mark`. There are 3 other places in this method that should be changed as well.